### DISCUSSION TITLE
Renovate recreate existing PRs when choose to group dependencies

### DISCUSSION THREAD

**Discussion by alaamub** [renovatebot/renovate#12237] [Request Help]
Hi, we use monorepo, and one of the configs I added is to group all python updates per directory, which includes both updating the `requirements.in` and `requirements.txt` and choose unique PR title for each update. However, every time I run renovate, it creates a new PR again for the closed groups even when none of the modules have been updated. I think I understand the logic here because renovate has no way to tell if an update exists already because it is not a unique key. but I was wondering if there is a solution for this, specifically, allowing PRs to be created with titles that include unique hash for the update or something like that. 

<details><summary>Click me to see config</summary>

```
{
  "repositories": ["org/monorepo"],
  "includePaths": ["project_group1/**"],
  "updateLockFiles": false,
  "lockFileMaintenance": {
    "enabled": false
  },
  "pip_requirements": {
    "enabled": true
  },
  "packageRules": [
 {
      "paths": ["project_group1/**"],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "vulnerable dependencies for project_group1",
    }
}
```
</details>

Also, I'm noticing that python `pip_requirements` always updates `requirements.txt`, how can I only tell it to update `requirements.in` only?

**Comment by rarkins** [comment]
> Hi, we use monorepo, and one of the configs I added is to group all python updates per directory, which includes both updating the `requirements.in` and `requirements.txt` and choose unique PR title for each update. However, every time I run renovate, it creates a new PR again for the closed groups even when none of the modules have been updated. I think I understand the logic here because renovate has no way to tell if an update exists already because it is not a unique key. but I was wondering if there is a solution for this, specifically, allowing PRs to be created with titles that include unique hash for the update or something like that.

No solution I'm aware of

> Also, I'm noticing that python `pip_requirements` always updates `requirements.txt`, how can I only tell it to update `requirements.in` only?

https://docs.renovatebot.com/modules/manager/#extending-a-managers-default-filematch

**Reply by alaamub** [reply]
Thanks!

**Reply by aira-xite** [reply]
@alaamub hi!
Have you solved the problem? If so, can you please post your solution?

**Comment by rarkins** [comment]
The new option recreateWhen=always may be useful here, but use it with caution (it means you can never close to ignore a PR)
